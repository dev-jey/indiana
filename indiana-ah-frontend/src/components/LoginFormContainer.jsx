import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Circle } from 'better-react-spinkit';
import { withRouter } from 'react-router-dom';
import { loginWithEmail } from '../redux/actions/authActions';
import InputField from './common/input/InputComponent.jsx';
import Button from '../styles/styledComponents/Button.jsx';
import { logo } from '../assets/images/svg';

export class LoginFormContainer extends Component {
  state = {
    data: {
      email: '',
      password: ''
    },
    error: ''
  };

  handleSubmit = async (e) => {
    e.preventDefault();
    await this.props.loginWithEmail(this.state.data, this.props);
    const { error } = this.props.auth;
    this.setState({ error });
  };

  handleChange = ({ target: input }) => {
    const data = { ...this.state.data };
    const field = input.name;
    data[field] = input.value;
    this.setState({ data });
  };

  render() {
    const {
      data: { password, email },
      error
    } = this.state;
    const {
      auth: { isLoading },
      displayForm
    } = this.props;
    return (
      <div className="signup-form-container">
        <h3 className="text-center mt-5">
          <img src={logo} alt="facebook logo" className="signup-title" />
          <hr />
        </h3>
        <form onSubmit={this.handleSubmit} className="signup-form">
          <InputField
            placeholder="Email"
            type="email"
            id="email1"
            name="email"
            value={email}
            handleChange={this.handleChange}
          />
          <InputField
            placeholder="Password"
            type="password"
            id="password"
            name="password"
            value={password}
            handleChange={this.handleChange}
          />
          {error && <div className="text-danger text-center">{error}</div>}
          <Button
            type="submit"
            disabled={isLoading}
            bgColor
            width={'95%'}
            margin={'auto'}
            height={'4.8rem'}
            sm>
            login
            {isLoading && (
              <span className="button-loading">
                <Circle color={'rgba(255,255,255,1)'} />
              </span>
            )}
          </Button>
        </form>
        <div className="signup-form">
          <div className="social">
            <ul>
              <li className="social-auth">Or Login Using:</li>
              <li>
                <a href="https://indiana-ah-staging.herokuapp.com/auth/facebook">
                  <i className="fab fa-lg fa-facebook-f" />
                </a>
              </li>
              <li>
                <a href="https://indiana-ah-staging.herokuapp.com/auth/twitter">
                  <i className="fab fa-lg fa-twitter" />
                </a>
              </li>
              <li>
              <a href="https://indiana-ah-staging.herokuapp.com/auth/google">
                  <i className="fab fa-lg fa-google" />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <p className="d-flex justify-content-center">
          Dont have an account?{' '}
          <a href="#" onClick={() => displayForm('register')} className="ml-2">
            Create one
          </a>
        </p>
        <div className="d-flex justify-content-center bg-light w-100 p-4 mt-3">
          <a href="#" onClick={() => displayForm('reset')}>Forgot password?</a>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { loginWithEmail }
)(withRouter(LoginFormContainer));

LoginFormContainer.propTypes = {
  auth: PropTypes.object,
  displayForm: PropTypes.func,
  loginWithEmail: PropTypes.func
};
