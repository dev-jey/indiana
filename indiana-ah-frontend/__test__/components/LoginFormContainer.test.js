import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { LoginFormContainer } from '../../src/components/LoginFormContainer.jsx';


const props = {
  auth: { isLoading: false },
  error: ''
};

describe('Login Form', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<LoginFormContainer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test the login container form', () => {
  let wrapper;
  let loginWithEmail;
  const displayForm = jest.fn();
  beforeEach(() => {
    loginWithEmail = jest.fn();
    wrapper = mount(
      <LoginFormContainer
        {...props}
        loginWithEmail={loginWithEmail}
        displayForm={displayForm}
      />
    );
  });
  it('should match snap shot', () => {
    const tree = renderer.create(<LoginFormContainer {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should simulate the handleSubmit', () => {
    wrapper.find('form').simulate('submit');
    const input1 = wrapper.find('input').at(0);
    const input2 = wrapper.find('input').at(1);
    input1.simulate('change', { target: { value: 'fafa@gmail.com' } });
    input2.simulate('change', { target: { value: 'fafasecret22' } });
    expect(loginWithEmail).toHaveBeenCalled();
  });
  it('should simulate the failure of handleSubmit', () => {
    props.auth.isLoading = true;
    wrapper.find('form').simulate('submit');
    const input1 = wrapper.find('input').at(0);
    const input2 = wrapper.find('input').at(1);
    wrapper.setState({ error: 'error logging in' });
    input1.simulate('change', { target: { value: 'fafa@gmail.com' } });
    input2.simulate('change', { target: { value: 'fafasecret33' } });
    expect(wrapper.state('error')).toEqual('error logging in');
  });
  it('should simulate clicking the link that opens the register form', () => {
    const aTag = wrapper.find('a').at(3);
    aTag.simulate('click');
    expect(displayForm).toHaveBeenCalled();
  });
  it('should simulate clicking the link that opens the reset password form', () => {
    const aTag = wrapper.find('a').at(4);
    aTag.simulate('click');
    expect(displayForm).toHaveBeenCalled();
  });
});
