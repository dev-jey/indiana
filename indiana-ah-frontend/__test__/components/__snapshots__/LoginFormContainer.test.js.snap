// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form should match snapshot 1`] = `
<div
  className="signup-form-container"
>
  <h3
    className="text-center mt-5"
  >
    <img
      alt="facebook logo"
      className="signup-title"
      src="test-file-stub"
    />
    <hr />
  </h3>
  <form
    className="signup-form"
    onSubmit={[Function]}
  >
    <div
      className="input-field"
    >
      <label
        htmlFor="email"
      />
      <input
        className="sc-bdVaJa doFRbm"
        id="email1"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
      <small
        className="text-danger ml-3 form-text"
      />
    </div>
    <div
      className="input-field"
    >
      <label
        htmlFor="password"
      />
      <input
        className="sc-bdVaJa doFRbm"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <small
        className="text-danger ml-3 form-text"
      />
    </div>
    
    <button
      className="sc-htpNat hpCSJ"
      disabled={false}
      height="4.8rem"
      type="submit"
      width="95%"
    >
      login
    </button>
  </form>
  <div
    className="signup-form"
  >
    <div
      className="social"
    >
      <ul>
        <li
          className="social-auth"
        >
          Or Login Using:
        </li>
        <li>
          <a
            href="https://indiana-ah-staging.herokuapp.com/auth/facebook"
          >
            <i
              className="fab fa-lg fa-facebook-f"
            />
          </a>
        </li>
        <li>
          <a
            href="https://indiana-ah-staging.herokuapp.com/auth/twitter"
          >
            <i
              className="fab fa-lg fa-twitter"
            />
          </a>
        </li>
        <li>
          <a
            href="https://indiana-ah-staging.herokuapp.com/auth/google"
          >
            <i
              className="fab fa-lg fa-google"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
  <p
    className="d-flex justify-content-center"
  >
    Dont have an account?
     
    <a
      className="ml-2"
      href="#"
      onClick={[Function]}
    >
      Create one
    </a>
  </p>
  <div
    className="d-flex justify-content-center bg-light w-100 p-4 mt-3"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      Forgot password?
    </a>
  </div>
</div>
`;

exports[`test the login container form should match snap shot 1`] = `
<div
  className="signup-form-container"
>
  <h3
    className="text-center mt-5"
  >
    <img
      alt="facebook logo"
      className="signup-title"
      src="test-file-stub"
    />
    <hr />
  </h3>
  <form
    className="signup-form"
    onSubmit={[Function]}
  >
    <div
      className="input-field"
    >
      <label
        htmlFor="email"
      />
      <input
        className="sc-bdVaJa doFRbm"
        id="email1"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
      <small
        className="text-danger ml-3 form-text"
      />
    </div>
    <div
      className="input-field"
    >
      <label
        htmlFor="password"
      />
      <input
        className="sc-bdVaJa doFRbm"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <small
        className="text-danger ml-3 form-text"
      />
    </div>
    
    <button
      className="sc-htpNat hpCSJ"
      disabled={false}
      height="4.8rem"
      type="submit"
      width="95%"
    >
      login
    </button>
  </form>
  <div
    className="signup-form"
  >
    <div
      className="social"
    >
      <ul>
        <li
          className="social-auth"
        >
          Or Login Using:
        </li>
        <li>
          <a
            href="https://indiana-ah-staging.herokuapp.com/auth/facebook"
          >
            <i
              className="fab fa-lg fa-facebook-f"
            />
          </a>
        </li>
        <li>
          <a
            href="https://indiana-ah-staging.herokuapp.com/auth/twitter"
          >
            <i
              className="fab fa-lg fa-twitter"
            />
          </a>
        </li>
        <li>
          <a
            href="https://indiana-ah-staging.herokuapp.com/auth/google"
          >
            <i
              className="fab fa-lg fa-google"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
  <p
    className="d-flex justify-content-center"
  >
    Dont have an account?
     
    <a
      className="ml-2"
      href="#"
      onClick={[Function]}
    >
      Create one
    </a>
  </p>
  <div
    className="d-flex justify-content-center bg-light w-100 p-4 mt-3"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      Forgot password?
    </a>
  </div>
</div>
`;
